packages/plugin-parser/src/ParsingEngine.ts

```diff
@@ -11,18 +11,21 @@ export interface Parser<T = {}> {
 
 export class ParsingEngine<T = {}> {
     static readonly id: ParsingIdentifier;
-    static defaultParser: Constructor<Parser>;
+    static readonly defaultParser: ParserConstructor;
+    static fallbacks?: ParsingIdentifier[];
     readonly editor: JWEditor;
     readonly parsers: Parser<T>[] = [];
     readonly parsingMap = new Map<T, VNode[]>();
 
     constructor(editor: JWEditor) {
         this.editor = editor;
-        const defaultParser = new this.constructor.defaultParser(this);
-        if (defaultParser.predicate) {
-            throw `Default renderer cannot have a predicate.`;
-        } else {
-            this.parsers.push(defaultParser);
+        if (this.constructor.defaultParser) {
```

Comment:
I don't understand this change. `ParsingEngines` *must* have a `defaultParser`, why the `if` ?